Python/pywarpx/fields.py

```diff
@@ -525,8 +526,11 @@ def __setitem__(self, index, value):
                 if isinstance(value, np.ndarray):
                     slice_value = value3d[global_slices]
                     if libwarpx.libwarpx_so.Config.have_gpu:
+                        xp, cupy_status = load_cupy()
```

Comment:
Two suggestions:
- move the load_copy outside of the loop, inside a `if libwarpx.libwarpx_so.Config.have_gpu:` to avoid spam
- check `cupy_status` and print immediately after, otherwise missing managed memory will crash before you see that the issue is a missing cupy

Reply:
Sounds good on avoiding the repeated prints. 
Just want to point out that currently it would print before crashing since it won't crash until line 534 😉